### PR TITLE
feat(evpn unit test): adding evpn mock unit test

### PR DESCRIPTION
**Pull request by atulpatel261194**
image

### CODE REVIEW

**Comment by glimchb** [comment]
please add line to https://github.com/opiproject/godpu/blob/main/Makefile#L29 to generate new mockery

**Review by glimchb — APPROVED** [review]
Lgtm

File: network/bridge_port_test.go [hunk 1]
```diff
@@ -3,4 +3,162 @@
...
+}
+
+func TestDeleteBridgePortError(t *testing.T) {
+	mockEvpnClient := &mocks.EvpnClient{}
+
+	expectedError := errors.New("mocked error")
+	mockEvpnClient.On("DeleteBridgePort", mock.Anything, mock.Anything, mock.Anything).
+		Return(nil, expectedError)
+
+	resultResponse, err := mockEvpnClient.DeleteBridgePort(context.TODO(), "bp1", true)
```
Comment (artek-koltun) [inline network/bridge_port_test.go:71]: @atulpatel261194 A bit late, but why do you test the generated mock here in these tests? Testing of mocks adds little value to our test suite.
I think we should test the implementation(evpnClientImpl) calls a corresponding method in pb.BridgePortServiceClient instead...

Comment (atulpatel261194) [reply network/bridge_port_test.go:71]: @artek-koltun testing the generated mocks protects the changes in .proto files.
i believe we can't test the real api's which create, delete and update different port, bridge, svi and vrf. if there is any way i can test these api's directly Please let me know.

Comment (artek-koltun) [reply network/bridge_port_test.go:71]: There is actually a way. You can use dependency injection to replace the real connection components. The test would look like
```Go
func TestCreateBridgePortSuccess(t *testing.T) {
	wantRequest := &_go.CreateBridgePortRequest{
		BridgePortId: "bp1",
		BridgePort: &pb.BridgePort{
			Spec: &pb.BridgePortSpec{
				MacAddress:     []byte("00:11:22:aa:bb:cc"),
				Ptype:          _go.BridgePortType_ACCESS,
				LogicalBridges: []string{"lb1", "lb2"},
			},
		},
	}
	wantResponse := wantRequest.BridgePort

	mockClient := mocks.NewBridgePortServiceClient(t)
	mockClient.EXPECT().CreateBridgePort(mock.Anything, wantRequest).
		Return(proto.Clone(wantResponse).(*pb.BridgePort), nil)

	connClosed := false
	mockConn := mocks.NewConnector(t)
	mockConn.EXPECT().NewConn().Return(
		&grpc.ClientConn{},
		func() { connClosed = true },
		nil,
	)

	c, _ := network.NewBridgePortWithArgs(
		mockConn,
		func(grpc.ClientConnInterface) pb.BridgePortServiceClient {
			return mockClient
		},
	)
	
	response, err := c.CreateBridgePort(
		context.Background(),
		"bp1", "00:11:22:aa:bb:cc", "access", []string{"lb1", "lb2"},
	)

	assert.NoError(t, err)
	assert.True(t, proto.Equal(response, wantResponse))
	assert.Equal(t, true, connClosed)
}
```

[Here](https://github.com/opiproject/godpu/blob/main/storage/nvme_subsystem_test.go) you can see how I tested subsystem with the table approach

**Pull request merged**